wordpress-mobile/WordPress-Android
# Remove unused Gutenberg Mobile feature flags

Removes unused Gutenberg Mobile feature flags. Similar to https://github.com/wordpress-mobile/WordPress-iOS/pull/17019, but for WPAndroid.

- `CONTACT_INFO_BLOCK_AVAILABLE`
- `LAYOUT_GRID_BLOCK_AVAILABLE`

To test:

- Smoke test each of the above features to make sure this PR didn't inadvertently break them.
- Make sure their feature no longer appears on the "Test feature configuration"/"Debug settings" screen (tap profile photo > App Settings)

## Regression Notes
1. Potential unintended areas of impact

If these feature flags are not removed properly, they can impact the features they were previously controlling.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Smoke test each feature to make sure it still works.

3. What automated tests I added (or what prevented me from doing so)

(**I'm not certain about this**) These feature flags affect blocks which all have tests in the Gutenberg Mobile repository; however, that doesn't help us verify that this PR didn't break those features because they are explicitly enabled in the demo app where tests are run.

PR submission checklist:
